internal/cluster/cluster.go

```diff
@@ -399,15 +400,22 @@ func (m *Manager) ClusterMonitoring() {
 					log.WithError(err).Errorf("failed to refresh cluster %s state", cluster.ID)
 					continue
 				}
-
 				if swag.StringValue(clusterAfterRefresh.Status) != swag.StringValue(cluster.Status) {
 					log.Infof("cluster %s updated status from %s to %s via monitor", cluster.ID,
 						swag.StringValue(cluster.Status), swag.StringValue(clusterAfterRefresh.Status))
 				}
-
 				if m.shouldTriggerLeaseTimeoutEvent(cluster, curMonitorInvokedAt) {
 					m.triggerLeaseTimeoutEvent(ctx, cluster)
 				}
+				if m.ocmClient != nil && m.ocmClient.Config.WithAMSSubscriptions {
+					if err := m.GetCredentials(cluster); err != nil {
+						continue
+					}
+					consoleUrl := fmt.Sprintf("%s.%s.%s", ConsoleUrlPrefix, cluster.Name, cluster.BaseDNSDomain)
```

Comment:
It's going to spam our logs.
Is it supposed to run on every stage? I think it would be better to check that only on Finalizing, no?

Reply:
> It's going to spam our logs.

What will "spam our logs"? the error printing to logs if `UpdateSubscriptionConsoleUrlIfNeeded` failed? this is no different than all other functionality in the cluster monitoring.

> Is it supposed to run on every stage? I think it would be better to check that only on Finalizing, no? as part of RefreshStatus function is the state machine

What is the difference?  `GetCredentials` will return nil only if we are in [`installing`, `finalizing`, `installed`] - only then the OCM call will be generated, and even then it will be done only once according to a status variable inside the inner function (the name of the function also explain that the OCM call is done only if needed.

Also, logically, it is not part of the RefreshStatus because there is nothing to refresh in the status - this is just sending a call to an external dependency.

Reply:
> > It's going to spam our logs.
> 
> What will "spam our logs"? the error printing to logs if `UpdateSubscriptionConsoleUrlIfNeeded` failed? this is no different than all other functionality in the cluster monitoring.
> 
> > Is it supposed to run on every stage? I think it would be better to check that only on Finalizing, no? as part of RefreshStatus function is the state machine
> 

Sorry,  I mistakenly thought that `fmt.Sprintf` is a log.

> What is the difference? `GetCredentials` will return nil only if we are in [`installing`, `finalizing`, `installed`] - only then the OCM call will be generated, and even then it will be done only once according to a status variable inside the inner function (the name of the function also explain that the OCM call is done only if needed.
> 
> Also, logically, it is not part of the RefreshStatus because there is nothing to refresh in the status - this is just sending a call to an external dependency.

But it does seem relevant that a code that is dependant on a state would need to be inside the state-machine.

Reply:
> But it does seem relevant that a code that is dependant on a state would need to be inside the state-machine

I am not sure I agree with that, clusterApi.RefreshStatus() is also called in some places inside inventory.go, we don't want to update AMS subscription (or at least try to) on each one of those calls.

Reply:
@filanov WDYT regarding the location of this logic?

Reply:
I think that all this block should be inside a function.
I see no reason to call it for all the states, this logic is relevant for the installation phase so it probably need to be in a transition.

Reply:
Doesn't it mean that it will run only once? @filanov ? 
What happens if it fails? we won't try again?

Reply:
There is no transition `preparing-for-installation` --> `installing`.
Moved it to `inventory.ClusterInstall` API.

Reply:
Yes this transition happens strangely in 
https://github.com/openshift/assisted-service/blob/59ccff6c37d15d7ef220443d500a594f4624af4f/internal/cluster/installer.go#L56-L59
But isn't the console check should be part of `RefreshStatus` of `Finalizing`?

Reply:
There is no "console check", there is only a method `clusterApi.GetCredentials(cluster)` that return `nil` if the status is [`installing`, `finallizing`, `installed`] or an error otherwise.

I am using it the same way it is used in https://github.com/openshift/assisted-service/blob/075ec70efd1aced697e35fdb3a991e0079349b33/internal/bminventory/inventory.go#L3262